src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -202,12 +203,58 @@ namespace levin
       return out;
     }
 
+    epee::levin::message_writer make_tx_inv_message(std::vector<crypto::hash>&& txs, const bool pad)
+    {
+      NOTIFY_TX_POOL_INV::request request{};
+      request.txs = std::move(txs);
+
+      if (pad)
+      {
+        size_t bytes = 9 /* header */ + 4 /* 1 + 'txs' */ + tools::get_varint_data(request.txs.size()).size();
+        for(auto tx_hash = request.txs.begin(); tx_hash!=request.txs.end(); ++tx_hash)
+          bytes += tools::get_varint_data(sizeof(*tx_hash)).size() + sizeof(*tx_hash);
```

Comment:
The serialized message stores all hashes in a single field, so this is incorrect.

```suggestion
        size_t bytes = 9 /* header */ + 4 /* 1 + 'txs' */ + tools::get_varint_data(request.txs.size() * 32).size() + request.txs.size() * 32;
```

Although even that is incorrect, it uses the consensus (COCOA BEANS) varint, not the epee varint and misses the field tag size. This is the same as the `make_tx_message` padding calculation. 

@vtnerd does this slight difference in calculation of the size vs the real size matter for the padding?

Reply:
This is why I promoted the usage of the technique the SSL PR used - it's more accurate, quicker, and easier to just pad the end of the levin message as the extra trailing data gets dropped. Arguably this feature should be dropped entirely (see below).

Assuming this implementation is kept, it's possibly leaking data, but it's unlikely to matter. This feature is only useful if the connection is encrypted, which currently occurs only over Tor/I2P, both of which are already padded to disrupt traffic shaping analysis. In other words, Tor/I2P already do what we want here.

When/if SSL gets merged, this should be re-thought as SSL provides no automatic padding, which is why I added randomized padding (at the levin "layer") in that patch. It pads a randomized amount up to 8KiB, but perhaps it should just "round up" like Tor/I2P do (if it works for those networks, surely it works for us).